src/pages/Collections/index.tsx

```diff
@@ -8,19 +8,21 @@ import CollectionsComponent from './components/CollectionsComponent';
 import SearchComponent from '../../components/SearchComponent';
 
 const CollectionsPage: FC = () => {
-  const [queryParams, setQueryParams] = useSearchParams();
+  const [queryParams] = useSearchParams();
   const [currentPage, setCurrentPage] = useState<number>(1);
-  const searchString = queryParams.get('search') || '';
+  const [searchString, setSearchString] = useState<string | undefined>(
+    queryParams.get('search') || '',
+  );
 
-  const onSearchChange = (value: string) => {
-    if (!value) {
-      queryParams.delete('search');
-    } else {
-      queryParams.set('search', value);
-    }
+  useEffect(() => {
```

Comment:
It looks like reusable hook. This code repeats 4 times in current request. Lets made it.